### DISCUSSION TITLE
Routing with a button (without using next/Link)

### DISCUSSION THREAD

**Discussion by YuriPadilla** [neuefische/questions#521] [Web] [answered]
Hello There!

I have a button and it does some action like: _handleSubmit_, or _handleChange, etc... And I want this button in addition to redirect to another page when clicking on it. 

I would just use the useRouter like this: 
```JS
import { useRouter } from "next/router";

export default function AnyPage() {
    const router = useRouter();

    function handleSomeAction() {
        //some code here for some action
        router.push("/AnotherPage");
    }

    return 
        <button type="button" onClick={handleSomeAction}>
            Do some action
        </button>;
}
```

I was told before that to redirect to another page, it is better to use the next/Link, and the way I exposed it above is somehow a _"bad practice"_, but in this case I want to do this redirection to another page after the action _"handleSomeAction"_ is completed... I mean... then inside this button.

I don't see how I can use next/Link here...

What should I do???
Is it correct my proposition???

Thanks a lot in advance to all of you for your answer! 😀 👍

Many greetings
Yuri

**Comment by RobinWitt** [comment, marked as answer]
Hey Yuri, using `router.push` at the end of a `handleSomethingFunction` should be alright, I have seen a lot of use cases for this. Just be aware of not confusing the user too much, maybe you can add a small delay with a message "redirecting to ..." so it is clear that the redirect is happening intentionally.
Hope this helps.

**Reply by NikRoe** [reply]
Hi there,

the use case you describe, Yuri, is a very good example when we want to route imperatively. If we only want a user to only be redirected if a certain criteria is met (like something else is to be executed before), we can work with `router.push()`.

Basically: what Robin said is absolutely right and you can also look at our [handout](https://github.com/neuefische/web-bo-22-2/blob/main/sessions/nextjs-dynamic-routes/nextjs-dynamic-routes.md#imperative-routing).

**Reply by YuriPadilla** [reply]
Hi Robin, hi Niklas!

Thank you guys for your answer. It was really helpful and I have a better understanding about redirecting.

Greetings!!! 😀 👍
